source/structured-cloneable.d.ts

```diff
@@ -0,0 +1,66 @@
+import type {TypedArray} from './typed-array';
+
+type StructuredCloneablePrimitive =
+	| string
+	| number
+	| bigint
+	| boolean
+	| null
+	| undefined
+	| Boolean
+	| Number
+	| String;
+
+type StructuredCloneableData =
+	| ArrayBuffer
+	| DataView
+	| Date
+	| Error
+	| RegExp
+	| TypedArray;
+
+type StructuredCloneableCollection =
+	| StructuredCloneable[]
+	| {[k: string]: StructuredCloneable; [k: number]: StructuredCloneable}
+	| Map<StructuredCloneable, StructuredCloneable>
+	| Set<StructuredCloneable>;
```

Comment:
I would use the read-only type versions here so it can support read-only collections too.